docs/source/developer/architecture/backend/data/GroupingMap.rst

```diff
@@ -0,0 +1,75 @@
+
+Usage:
+======
+At any point after a state has been initialized, the state's grouping-schema map may be accessed using:
+
+1. Import the necessary classes:
+
+.. code-block:: python
+
+   from from snapred.backend.dao.state import GroupingMap
+   from snapred.backend.data.DataFactoryService import DataFactoryService
+
+2. Access the state's grouping map:
+
+.. code-block:: python
+
+   runNumber = "52280"
+   dataService = DataFactoryService()
+   groupingMap = dataService.getGroupingMap(runNumber)
+
+3. Obtain the filename for the "column" (lite mode) grouping schema:
+
+.. code-block:: python
+
+   groupingScheme = "column"
+   useLiteMode = True
+
+   # A relative path here will refer to <instrument.calibration.powder.grouping.home>
+   columnFilename = groupingMap.getMap(useLiteMode)[groupingScheme].definition
+
+
+Initialization specifics:
+=========================
+
+The template file ``defaultGroupingMap.json`` must be filled-in and placed "by hand" at <instrument.calibration.powder.grouping.home> by the CIS.
+
+The creation of the ``groupingMap.json`` file located at the state-root directory is triggered at two distinct code locations, whenever a state-root directory doesn't exist.  These trigger locations are at: 1) ``LocalDataService.readStateConfig``, and (2) ``LocalDataService.initializeState``.  Once the ``groupingMap.json`` file has been created, as it may contain user-specified information, it will never be overwritten.
+
+The ``GroupingMap`` data object is an attribute of the ``StateConfig`` data object, however it is represented on disk as a separate JSON file.  In many cases, the ``GroupingMap`` is required when the complete ``StateConfig`` is not.  For this reason, it is generally accessed using ``LocalDataService.readGroupingMap`` (see the "defects" section below), rather than ``LocalDataService.readStateConfig``.  This is why two separate initialization paths have been implemented.  Generally, the ``initializeState`` path is used by the application code and unit-test code, whereas the ``readStateConfig`` path at present only seems to be used by the unit-test code.
+
+Validation specifics:
+=====================
+
+:term:`Pydantic` validators are used to validate the ``GroupingMap`` instance at point of loading.  Any grouping-schema files which do not exist (or have an incorrect format) are not loaded to the resident ``GroupingMap`` -- in these cases, warnings will be logged but the loading will be allowed to continue.  When the ``GroupingMap`` is loaded using the ``readStateConfig`` path, the validity of the ``GroupingMap.stateId`` is checked to ensure that it corresponds to that of the state itself.
+
+Potential defects:
```

Comment:
I appreciate the potential defects section, and all the additional glossary entries!